File: binding/include/libqbuttongroup.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qbuttongroup.html
pub const qbuttongroup = struct {

    /// New constructs a new QButtonGroup object.
    pub fn New() QtC.QButtonGroup {
        return qtc.QButtonGroup_new();
    }


    /// New2 constructs a new QButtonGroup object.
    pub fn New2(parent: ?*anyopaque) QtC.QButtonGroup {
        return qtc.QButtonGroup_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QButtonGroup_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbuttongroup.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetExclusive(self: ?*anyopaque, exclusive: bool) void {
        qtc.QButtonGroup_SetExclusive(@ptrCast(self), exclusive);
    }

    pub fn Exclusive(self: ?*anyopaque, ) bool {
        return qtc.QButtonGroup_Exclusive(@ptrCast(self));
    }

    pub fn AddButton(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QButtonGroup_AddButton(@ptrCast(self), @ptrCast(param1));
    }

    pub fn RemoveButton(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QButtonGroup_RemoveButton(@ptrCast(self), @ptrCast(param1));
    }

    pub fn Buttons(self: ?*anyopaque, ) []const u8 {
        return qtc.QButtonGroup_Buttons(@ptrCast(self));
    }

    pub fn CheckedButton(self: ?*anyopaque, ) QtC.QAbstractButton {
        return qtc.QButtonGroup_CheckedButton(@ptrCast(self));
    }

    pub fn Button(self: ?*anyopaque, id: i32) QtC.QAbstractButton {
        return qtc.QButtonGroup_Button(@ptrCast(self), id);
    }

    pub fn SetId(self: ?*anyopaque, button: ?*anyopaque, id: i32) void {
        qtc.QButtonGroup_SetId(@ptrCast(self), @ptrCast(button), id);
    }

    pub fn Id(self: ?*anyopaque, button: ?*anyopaque) i32 {
        return qtc.QButtonGroup_Id(@ptrCast(self), @ptrCast(button));
    }

    pub fn CheckedId(self: ?*anyopaque, ) i32 {
        return qtc.QButtonGroup_CheckedId(@ptrCast(self));
    }

    pub fn ButtonClicked(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QButtonGroup_ButtonClicked(@ptrCast(self), @ptrCast(param1));
    }

    pub fn OnButtonClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QButtonGroup_Connect_ButtonClicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ButtonPressed(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QButtonGroup_ButtonPressed(@ptrCast(self), @ptrCast(param1));
    }

    pub fn OnButtonPressed(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QButtonGroup_Connect_ButtonPressed(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ButtonReleased(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QButtonGroup_ButtonReleased(@ptrCast(self), @ptrCast(param1));
    }

    pub fn OnButtonReleased(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QButtonGroup_Connect_ButtonReleased(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ButtonToggled(self: ?*anyopaque, param1: ?*anyopaque, param2: bool) void {
        qtc.QButtonGroup_ButtonToggled(@ptrCast(self), @ptrCast(param1), param2);
    }

    pub fn OnButtonToggled(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, bool) callconv(.c) void) void {
        qtc.QButtonGroup_Connect_ButtonToggled(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn IdClicked(self: ?*anyopaque, param1: i32) void {
        qtc.QButtonGroup_IdClicked(@ptrCast(self), param1);
    }

    pub fn OnIdClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QButtonGroup_Connect_IdClicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn IdPressed(self: ?*anyopaque, param1: i32) void {
        qtc.QButtonGroup_IdPressed(@ptrCast(self), param1);
    }

    pub fn OnIdPressed(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QButtonGroup_Connect_IdPressed(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn IdReleased(self: ?*anyopaque, param1: i32) void {
        qtc.QButtonGroup_IdReleased(@ptrCast(self), param1);
    }

    pub fn OnIdReleased(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QButtonGroup_Connect_IdReleased(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn IdToggled(self: ?*anyopaque, param1: i32, param2: bool) void {
        qtc.QButtonGroup_IdToggled(@ptrCast(self), param1, param2);
    }

    pub fn OnIdToggled(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, bool) callconv(.c) void) void {
        qtc.QButtonGroup_Connect_IdToggled(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QButtonGroup_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbuttongroup.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QButtonGroup_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbuttongroup.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn AddButton2(self: ?*anyopaque, param1: ?*anyopaque, id: i32) void {
        qtc.QButtonGroup_AddButton2(@ptrCast(self), @ptrCast(param1), id);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QButtonGroup_Delete(@ptrCast(self));
    }
};

File: binding/src/libqbuttongroup.cpp
#include <QAbstractButton>
#include <QButtonGroup>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qbuttongroup.h>
#include "libqbuttongroup.h"
#include "libqbuttongroup.hxx"

QButtonGroup* QButtonGroup_new() {
	 return new QButtonGroup();
}

QButtonGroup* QButtonGroup_new2(QObject* parent) {
	 return new QButtonGroup(parent);
}

libqt_string QButtonGroup_Tr(const char* s) {
	QString _ret = QButtonGroup::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QButtonGroup_SetExclusive(QButtonGroup* self, bool exclusive) {
	self->setExclusive(exclusive);
}

bool QButtonGroup_Exclusive(const QButtonGroup* self) {
	return self->exclusive();
}

void QButtonGroup_AddButton(QButtonGroup* self, QAbstractButton* param1) {
	self->addButton(param1);
}

void QButtonGroup_RemoveButton(QButtonGroup* self, QAbstractButton* param1) {
	self->removeButton(param1);
}

libqt_list QButtonGroup_Buttons(const QButtonGroup* self) {
	auto _ret = self->buttons();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		_data[_i] = _ret[_i];
	}
	return _arr;
}

QAbstractButton* QButtonGroup_CheckedButton(const QButtonGroup* self) {
	return self->checkedButton();
}

QAbstractButton* QButtonGroup_Button(const QButtonGroup* self, int id) {
	return self->button(id);
}

void QButtonGroup_SetId(QButtonGroup* self, QAbstractButton* button, int id) {
	self->setId(button, id);
}

int QButtonGroup_Id(const QButtonGroup* self, QAbstractButton* button) {
	return self->id(button);
}

int QButtonGroup_CheckedId(const QButtonGroup* self) {
	return self->checkedId();
}

void QButtonGroup_ButtonClicked(QButtonGroup* self, QAbstractButton* param1) {
	self->buttonClicked(param1);
}

void QButtonGroup_Connect_ButtonClicked(QButtonGroup* self, intptr_t slot) {
    void (*slotFunc)(QButtonGroup*, QAbstractButton*) = reinterpret_cast<void (*)(QButtonGroup*, QAbstractButton*)>(slot);
    QButtonGroup::connect(self, &QButtonGroup::buttonClicked, [self, slotFunc](QAbstractButton* param1) {
	slotFunc(self, param1);
    });
}

void QButtonGroup_ButtonPressed(QButtonGroup* self, QAbstractButton* param1) {
	self->buttonPressed(param1);
}

void QButtonGroup_Connect_ButtonPressed(QButtonGroup* self, intptr_t slot) {
    void (*slotFunc)(QButtonGroup*, QAbstractButton*) = reinterpret_cast<void (*)(QButtonGroup*, QAbstractButton*)>(slot);
    QButtonGroup::connect(self, &QButtonGroup::buttonPressed, [self, slotFunc](QAbstractButton* param1) {
	slotFunc(self, param1);
    });
}

void QButtonGroup_ButtonReleased(QButtonGroup* self, QAbstractButton* param1) {
	self->buttonReleased(param1);
}

void QButtonGroup_Connect_ButtonReleased(QButtonGroup* self, intptr_t slot) {
    void (*slotFunc)(QButtonGroup*, QAbstractButton*) = reinterpret_cast<void (*)(QButtonGroup*, QAbstractButton*)>(slot);
    QButtonGroup::connect(self, &QButtonGroup::buttonReleased, [self, slotFunc](QAbstractButton* param1) {
	slotFunc(self, param1);
    });
}

void QButtonGroup_ButtonToggled(QButtonGroup* self, QAbstractButton* param1, bool param2) {
	self->buttonToggled(param1, param2);
}

void QButtonGroup_Connect_ButtonToggled(QButtonGroup* self, intptr_t slot) {
    void (*slotFunc)(QButtonGroup*, QAbstractButton*, bool) = reinterpret_cast<void (*)(QButtonGroup*, QAbstractButton*, bool)>(slot);
    QButtonGroup::connect(self, &QButtonGroup::buttonToggled, [self, slotFunc](QAbstractButton* param1, bool param2) {
	slotFunc(self, param1, param2);
    });
}

void QButtonGroup_IdClicked(QButtonGroup* self, int param1) {
	self->idClicked(param1);
}

void QButtonGroup_Connect_IdClicked(QButtonGroup* self, intptr_t slot) {
    void (*slotFunc)(QButtonGroup*, int) = reinterpret_cast<void (*)(QButtonGroup*, int)>(slot);
    QButtonGroup::connect(self, &QButtonGroup::idClicked, [self, slotFunc](int param1) {
	slotFunc(self, param1);
    });
}

void QButtonGroup_IdPressed(QButtonGroup* self, int param1) {
	self->idPressed(param1);
}

void QButtonGroup_Connect_IdPressed(QButtonGroup* self, intptr_t slot) {
    void (*slotFunc)(QButtonGroup*, int) = reinterpret_cast<void (*)(QButtonGroup*, int)>(slot);
    QButtonGroup::connect(self, &QButtonGroup::idPressed, [self, slotFunc](int param1) {
	slotFunc(self, param1);
    });
}

void QButtonGroup_IdReleased(QButtonGroup* self, int param1) {
	self->idReleased(param1);
}

void QButtonGroup_Connect_IdReleased(QButtonGroup* self, intptr_t slot) {
    void (*slotFunc)(QButtonGroup*, int) = reinterpret_cast<void (*)(QButtonGroup*, int)>(slot);
    QButtonGroup::connect(self, &QButtonGroup::idReleased, [self, slotFunc](int param1) {
	slotFunc(self, param1);
    });
}

void QButtonGroup_IdToggled(QButtonGroup* self, int param1, bool param2) {
	self->idToggled(param1, param2);
}

void QButtonGroup_Connect_IdToggled(QButtonGroup* self, intptr_t slot) {
    void (*slotFunc)(QButtonGroup*, int, bool) = reinterpret_cast<void (*)(QButtonGroup*, int, bool)>(slot);
    QButtonGroup::connect(self, &QButtonGroup::idToggled, [self, slotFunc](int param1, bool param2) {
	slotFunc(self, param1, param2);
    });
}

libqt_string QButtonGroup_Tr2(const char* s, const char* c) {
	QString _ret = QButtonGroup::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QButtonGroup_Tr3(const char* s, const char* c, int n) {
	QString _ret = QButtonGroup::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QButtonGroup_AddButton2(QButtonGroup* self, QAbstractButton* param1, int id) {
	self->addButton(param1, id);
}

void QButtonGroup_Delete(QButtonGroup* self) {
    delete self;
}

